File: Sorting/Bubble_Sort's/Bubble.java
import java.util.Arrays;
import java.util.Scanner;

public class Bubble
{
    private static void bubble_sort(int[] arr)
    {
        int n = arr.length;
        for(int i = n-1 ;i>=1;i--)
        {
            int didSwap = 0;
            for(int j = 0 ;j<=i-1;j++)
            {
                if(arr[j]>arr[j+1])
                {
                    int temp = arr[j+1];
                    arr[j+1] = arr[j];
                    arr[j] = temp;
                    didSwap = 1;
                }
            }
            if(didSwap == 0)
            {
                return;
            }
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the number of elements: ");
        int n = sc.nextInt();

        int[] arr = new int[n];

        System.out.println("Enter the elements:");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }

        bubble_sort(arr);
        System.out.println("Sorted array: " + Arrays.toString(arr));

        sc.close(); 
    }
}

File: Sorting/Bubble_Sort's/Bubble.js
function bubble_Sort(arr)
{
    let n = arr.length;
    for(let i = n-1;i>=1;i--)
    {
        let didSwap = false;
        for(let j = 0 ;j<=i-1;j++)
        {
            if(arr[j]>arr[j+1])
            {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                didSwap = true;
            }
        }
        if(didSwap==false)
        {
            break;
        }
    }
    return arr;
}

let arr = [13,46,24,52,20,9]
console.log(bubble_Sort(arr));
